ee/hogai/assistant/base.py

```diff
@@ -322,29 +306,12 @@ async def _init_or_update_state(self):
     async def _process_update(self, update: Any) -> list[AssistantResultUnion] | None:
         update = extract_stream_update(update)
 
-        new_message: AssistantResultUnion | None = None
         if not isinstance(update, AssistantDispatcherEvent):
-            if is_state_update(update):
-                _, new_state = update
-                self._state = validate_state_update(new_state, self._state_type)
-            elif is_value_update(update) and (new_message := await self._aprocess_value_update(update)):
-                return [new_message]
-
-            if is_message_update(update):
-                # Convert the message chunk update to a dispatcher event to prepare for a bright future without LangGraph
-                message, state = update[1]
-                if not isinstance(message, AIMessageChunk):
-                    return None
-                update = AssistantDispatcherEvent(
-                    action=MessageChunkAction(message=message), node_name=state["langgraph_node"]
-                )
-
-        if isinstance(update, AssistantDispatcherEvent) and (new_message := self._stream_processor.process(update)):
-            return [new_message] if new_message else None
-
-        return None
+            if updates := self._stream_processor.process_langgraph_update(LangGraphUpdateEvent(update=update)):
```

Comment:
Question: Is a case "update is handled, but there are no messages to emit" possible? Like an empty list return.